### PR TITLE
Fix a couple crashes

### PR DESCRIPTION
**Pull request by marc-hb**
See commit messages.

### CODE REVIEW

File: src/rimage.c [hunk 1]
```diff
@@ -86,11 +86,12 @@ int main(int argc, char *argv[])
 			usage(argv[0]);
 			break;
 		default:
+		 /* getopt's default error message is good enough */
 			break;
 		}
 	}
 
-	elf_argc = optind;
+	const int first_non_opt = optind;
```
Comment (ktrzcinx) [inline src/rimage.c:94]: Everywhere in project we declare new variables at the begin of function, I guess we shouldn't make here exception and keep code style consistent. Of course, then const must be removed, but it's not so critical for local variables.

**Review by lgirdwood — COMMENTED** [review]
@marc-hb thanks for these fixes, we can apply once code style issue is resolved. I will probably tag this as v1.6.

File: src/rimage.c (hunk 1 above)
Comment (marc-hb) [reply src/rimage.c:94]: This is a not just a code style question, this is an important new feature added by C99 to reduce the chance of memory corruption. Uninitialized variables is a cause of crashes and security issues.

I understand xt-xcc doesn't support C99 so it probably requires declarations at the start of a block (not function) but do we really need to keep user space tools stuck in the past too? C99 is 20 years old now.

Comment (lgirdwood) [reply src/rimage.c:94]: @marc-hb can you update this now, we can discuss this after v1.6. Thanks

Comment (marc-hb) [reply src/rimage.c]: Done.

**Comment by marc-hb** [comment]
submodule update submitted at https://github.com/thesofproject/sof/pull/3457

**Pull request merged**